[Bug]: AMM fuzzing panic: decimal division by 0

### Problem encountered

overnight fuzzing run against develop
```
2024-09-02T00:30:34.888Z	INFO	core	node/node.go:99	starting vega	{"version": "v0.78.0-dev", "commit-hash": "643133041a5cc06dab08e201cf0bf946d793e24b"}
```
hits panic
```
panic: decimal division by 0

goroutine 713230 [running]:
github.com/shopspring/decimal.Decimal.QuoRem({0xc00408c6a0, 0xfffffff1}, {0xc00408c6e0, 0xfffffff0}, 0x10)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:590 +0x2bc
github.com/shopspring/decimal.Decimal.DivRound({0xc00408c6a0?, 0x0?}, {0xc00408c6e0?, 0xfffffff0?}, 0x10)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:632 +0x4d
github.com/shopspring/decimal.Decimal.Div(...)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:577
code.vegaprotocol.io/vega/core/execution/amm.impliedPosition({0xc00408c620?, 0x0?}, {0x0?, 0x33f9500?}, {0xc0004914c0?, 0x581ebf?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/pool.go:544 +0x9f
code.vegaprotocol.io/vega/core/execution/amm.(*curve).positionAtPrice(0xc00b4e5440, 0x0?, 0x0?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/pool.go:73 +0x3f
code.vegaprotocol.io/vega/core/execution/amm.(*shapeMaker).makeBoundaryOrder(0xc00ea8a090, 0xc00e90bd20, 0x0?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/shape.go:97 +0xd1
code.vegaprotocol.io/vega/core/execution/amm.(*shapeMaker).calculateBoundaryOrders(0xc00ea8a090)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/shape.go:132 +0x1a7
code.vegaprotocol.io/vega/core/execution/amm.(*shapeMaker).makeShape(0xc00ea8a090)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/shape.go:361 +0x3f
code.vegaprotocol.io/vega/core/execution/amm.(*Pool).OrderbookShape(0xc00803f880?, 0xc005168660?, 0x1?, 0x6?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/shape.go:396 +0x2a
code.vegaprotocol.io/vega/core/execution/common.(*MarketLiquidity).updateAMMCommitment(0xc00a768500, 0x99)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/common/liquidity_provision_fees.go:162 +0x574
code.vegaprotocol.io/vega/core/execution/common.(*MarketLiquidity).OnTick(0xc00a768500, {0x486f138, 0xc004cc2720}, {0x0?, 0xedbec1620?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/common/liquidity_provision.go:390 +0x1fd
code.vegaprotocol.io/vega/core/execution/future.(*Market).OnTick(0xc008cdf500, {0x486f138, 0xc004cc2720}, {0x40?, 0xc00cd45f80?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:1087 +0x665
code.vegaprotocol.io/vega/core/execution.(*Engine).OnTick(0xc000f8bc00, {0x486f138, 0xc004cc2720}, {0x1?, 0x488ef60?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/engine.go:1223 +0x4f1
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/vegatime/service.go:111
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0xc00108e400, {0x486f138, 0xc004cc2720}, {0x40?, 0xc005b11040?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/vegatime/service.go:80 +0x263
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(0xc000906700, 0x1de8, {0xc008e56c40, 0x40}, {0xc000ff5b00?, 0x453c69?, 0x0?}, {0x0, 0x0}, {0xc005b11040, ...})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:1347 +0x976
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0xc0006e6cc0, {0xb8?, 0x10000000000?}, 0xc008e4bd40)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/abci/abci.go:168 +0x449
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0xc00da3b7d0?, {0x486f100?, 0x6e29a80?}, 0xc009658b40?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x25
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0xc000dafc20)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/nullchain.go:212 +0x556
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0xc000dafc20, 0x3b9aca00)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/nullchain.go:289 +0x1ce
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/server.go:132 +0x25
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 713228
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/server.go:131 +0x22f
```

[vega-sim-ddh9ag5d.zip](https://github.com/user-attachments/files/16836207/vega-sim-ddh9ag5d.zip)


### Observed behaviour

panic

### Expected behaviour

no panic

### Steps to reproduce

```bash
replay
```


### Software version

643133041a5cc06dab08e201cf0bf946d793e24b

### Failing test

vega-market-sim fuzz test

### Jenkins run

https://jenkins.vega.rocks/job/common/job/vega-market-sim-fuzzing-develop/5/artifact/network_home/vega-sim-ddh9ag5d/

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "0000000000000000000000000000000000000000000000000000000000000000": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "quantum": "1",
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "blockchains.evmBridgeConfigs": "{\"configs\":[{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }]}",
      "blockchains.ethereumRpcAndEvmCompatDataSourcesConfig": "{\"configs\": []}",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeRebateProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeRebateProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeRebateProgram.minClose": "1s",
      "governance.proposal.VolumeRebateProgram.minEnact": "1s",
      "governance.proposal.VolumeRebateProgram.minProposerBalance": "1",
      "governance.proposal.VolumeRebateProgram.minVoterBalance": "1",
      "governance.proposal.VolumeRebateProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeRebateProgram.requiredParticipation": "0.00001",
      "governance.proposal.transfer.maxClose": "8760h0m0s",
      "governance.proposal.transfer.maxEnact": "8760h0m0s",
      "governance.proposal.transfer.minClose": "1s",
      "governance.proposal.transfer.minEnact": "1s",
      "governance.proposal.transfer.minProposerBalance": "1",
      "governance.proposal.transfer.minVoterBalance": "1",
      "governance.proposal.transfer.requiredMajority": "0.5",
      "governance.proposal.transfer.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "limits.markets.proposePerpetualEnabled": "1",
      "limits.markets.ammPoolEnabled": "1",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0005",
      "market.fee.factors.makerFee": "0.0002",
      "market.fee.factors.treasuryFee": "0.0001",
      "market.fee.factors.buybackFee": "0.0001",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "0000000000000000000000000000000000000000000000000000000000000000",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "1",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "spam.order.minimumMarginQuantumMultiple": "0",
      "spam.order.minimumHoldingQuantumMultiple": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "0",
      "transfer.fee.maxQuantumAmount": "100",
      "transfer.feeDiscountMinimumTrackedAmount": "0.001",
      "transfer.feeDiscountDecayFraction": "0.5",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1",
      "market.aggressiveOrderBlockDelay": "1",
      "market.amm.minCommitmentQuantum": "1",
      "market.liquidity.maxAmmCalculationLevels": "100"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: decimal division by 0

goroutine 713230 [running]:
github.com/shopspring/decimal.Decimal.QuoRem({0xc00408c6a0, 0xfffffff1}, {0xc00408c6e0, 0xfffffff0}, 0x10)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:590 +0x2bc
github.com/shopspring/decimal.Decimal.DivRound({0xc00408c6a0?, 0x0?}, {0xc00408c6e0?, 0xfffffff0?}, 0x10)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:632 +0x4d
github.com/shopspring/decimal.Decimal.Div(...)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:577
code.vegaprotocol.io/vega/core/execution/amm.impliedPosition({0xc00408c620?, 0x0?}, {0x0?, 0x33f9500?}, {0xc0004914c0?, 0x581ebf?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/pool.go:544 +0x9f
code.vegaprotocol.io/vega/core/execution/amm.(*curve).positionAtPrice(0xc00b4e5440, 0x0?, 0x0?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/pool.go:73 +0x3f
code.vegaprotocol.io/vega/core/execution/amm.(*shapeMaker).makeBoundaryOrder(0xc00ea8a090, 0xc00e90bd20, 0x0?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/shape.go:97 +0xd1
code.vegaprotocol.io/vega/core/execution/amm.(*shapeMaker).calculateBoundaryOrders(0xc00ea8a090)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/shape.go:132 +0x1a7
code.vegaprotocol.io/vega/core/execution/amm.(*shapeMaker).makeShape(0xc00ea8a090)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/shape.go:361 +0x3f
code.vegaprotocol.io/vega/core/execution/amm.(*Pool).OrderbookShape(0xc00803f880?, 0xc005168660?, 0x1?, 0x6?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/amm/shape.go:396 +0x2a
code.vegaprotocol.io/vega/core/execution/common.(*MarketLiquidity).updateAMMCommitment(0xc00a768500, 0x99)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/common/liquidity_provision_fees.go:162 +0x574
code.vegaprotocol.io/vega/core/execution/common.(*MarketLiquidity).OnTick(0xc00a768500, {0x486f138, 0xc004cc2720}, {0x0?, 0xedbec1620?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/common/liquidity_provision.go:390 +0x1fd
code.vegaprotocol.io/vega/core/execution/future.(*Market).OnTick(0xc008cdf500, {0x486f138, 0xc004cc2720}, {0x40?, 0xc00cd45f80?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:1087 +0x665
code.vegaprotocol.io/vega/core/execution.(*Engine).OnTick(0xc000f8bc00, {0x486f138, 0xc004cc2720}, {0x1?, 0x488ef60?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/engine.go:1223 +0x4f1
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/vegatime/service.go:111
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0xc00108e400, {0x486f138, 0xc004cc2720}, {0x40?, 0xc005b11040?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/vegatime/service.go:80 +0x263
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(0xc000906700, 0x1de8, {0xc008e56c40, 0x40}, {0xc000ff5b00?, 0x453c69?, 0x0?}, {0x0, 0x0}, {0xc005b11040, ...})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:1347 +0x976
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0xc0006e6cc0, {0xb8?, 0x10000000000?}, 0xc008e4bd40)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/abci/abci.go:168 +0x449
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0xc00da3b7d0?, {0x486f100?, 0x6e29a80?}, 0xc009658b40?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x25
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0xc000dafc20)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/nullchain.go:212 +0x556
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0xc000dafc20, 0x3b9aca00)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/nullchain.go:289 +0x1ce
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/server.go:132 +0x25
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 713228
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/server.go:131 +0x22f
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -36,6 +36,7 @@
 - [11521](https://github.com/vegaprotocol/vega/issues/11521) - Restore `AMM` position factor when loading from a snapshot.
 - [11526](https://github.com/vegaprotocol/vega/issues/11526) - `EstimateAMMBounds` now respects the market's decimal places.
 - [11486](https://github.com/vegaprotocol/vega/issues/11486) - `AMMs` can now be submitted on markets with more decimal places than asset decimal places.
+- [11635](https://github.com/vegaprotocol/vega/issues/11635) - Handle expansion of one sided `AMMs` that reduce to point expansion when calculating order book shape.
 - [11561](https://github.com/vegaprotocol/vega/issues/11561) - Failing amends on `AMMs` now restore original properly.
 - [11583](https://github.com/vegaprotocol/vega/issues/11583) - Remove `AMMs` entirely from engine when market closes.
 - [11568](https://github.com/vegaprotocol/vega/issues/11568) - order book shape on closing `AMM` no longer panics.

--- a/core/execution/amm/shape.go
+++ b/core/execution/amm/shape.go
@@ -321,13 +321,15 @@ func (sm *shapeMaker) adjustRegion() bool {
 		return false
 	}
 
-	if sm.from.EQ(sm.to) && sm.from.EQ(sm.fairPrice) {
-		return false
-	}
-
 	// cap the range to the pool's bounds, there will be no orders outside of this
 	from := num.Max(sm.from, lower)
 	to := num.Min(sm.to, upper)
+
+	// expansion is a point region *at* fair-price, there are no orders
+	if from.EQ(to) && from.EQ(sm.fairPrice) {
+		return false
+	}
+
 	switch {
 	case sm.from.GT(sm.fairPrice):
 		// if we are expanding entirely in the sell range to calculate the order at price `from`

--- a/core/execution/amm/shape_test.go
+++ b/core/execution/amm/shape_test.go
@@ -36,6 +36,7 @@ func TestOrderbookShape(t *testing.T) {
 	t.Run("test orderbook shape boundary order when approx", testOrderbookShapeBoundaryOrder)
 	t.Run("test orderbook shape region not divisible by tick", testOrderbookSubTick)
 	t.Run("test orderbook shape closing pool close to base", testClosingCloseToBase)
+	t.Run("test orderbook shape point expansion at fair price", testPointExpansionAtFairPrice)
 }
 
 func testOrderbookShapeZeroPosition(t *testing.T) {
@@ -413,6 +414,39 @@ func testClosingCloseToBase(t *testing.T) {
 	buys, sells = p.pool.OrderbookShape(from, to, nil)
 
 	// should have one sell of volume 1
+	assert.Equal(t, 0, len(buys))
+	assert.Equal(t, 0, len(sells))
+}
+
+func testPointExpansionAtFairPrice(t *testing.T) {
+	p := newTestPoolWithRanges(t, num.NewUint(7), num.NewUint(10), num.NewUint(13))
+	defer p.ctrl.Finish()
+
+	base := p.submission.Parameters.Base
+
+	// range [10, 10] fair price is 10, no orders
+	ensurePositionN(t, p.pos, 0, num.UintZero(), 2)
+	buys, sells := p.pool.OrderbookShape(base, base, nil)
+	assert.Equal(t, 0, len(buys))
+	assert.Equal(t, 0, len(sells))
+
+	// now try with a one sided curve where the input range shrinks to a point-expansion
+	p = newTestPoolWithRanges(t, num.NewUint(7), num.NewUint(10), nil)
+	defer p.ctrl.Finish()
+
+	// range [10, 1000] but sell curve is empty so effective range is [10, 10] at fair-price
+	ensurePositionN(t, p.pos, 0, num.UintZero(), 2)
+	buys, sells = p.pool.OrderbookShape(base, num.NewUint(1000), nil)
+	assert.Equal(t, 0, len(buys))
+	assert.Equal(t, 0, len(sells))
+
+	// now try with a one sided curve where the input range shrinks to a point-expansion
+	p = newTestPoolWithRanges(t, nil, num.NewUint(10), num.NewUint(13))
+	defer p.ctrl.Finish()
+
+	// range [1, 10] but buy curve is empty so effective range is [10, 10] at fair-price
+	ensurePositionN(t, p.pos, 0, num.UintZero(), 2)
+	buys, sells = p.pool.OrderbookShape(num.NewUint(1), base, nil)
 	assert.Equal(t, 0, len(buys))
 	assert.Equal(t, 0, len(sells))
 }

